web/fido_app/webauthn_routes.py

```diff
@@ -109,32 +114,50 @@ def verify_registration_credentials():
         flash('Credential ID already exists.', 'error')
         return make_response(jsonify({'redirect': url_for('register')}), 401)
 
-    # Ensure the user's email isn't already in use (TODO: refactor into a function for
-    # both registration stages)
-    if User.query.filter_by(email=email).first():
-        flash('Email address is already in use', 'error')
-        return make_response(jsonify({'redirect': url_for('register')}), 401)
-
-    # Create a new user
-    user = User(
-        email=email,
-        ukey=ukey,
-        display_name=display_name,
-        public_key=bytes_to_base64url(verified_registration.credential_public_key),
-        credential_id=credential_id,
-        sign_count=verified_registration.sign_count,
-        authenticator_id=verified_registration.aaguid,
-        attestation_format=verified_registration.fmt,
-        user_verified=verified_registration.user_verified
-    )
-    db.session.add(user)
-    db.session.commit()
+    # If this is an existing user, update the db entry for that user
+    if current_user.is_authenticated:
+        existing_user = User.query.filter_by(email=email).first()
```

Comment:
I believe that `current_user` exposes the full user object (so you wouldn't have to call `User.query()`). Up to you if you want to change this.

Reply:
Thanks for pointing this out. I'll go ahead and change it.